src/chat/CustomerSupport/ProductManualService.cs

```diff
@@ -0,0 +1,45 @@
+using System.Net.Http.Headers;
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.VectorData;
+
+public class ProductManualService
+{
+    private readonly IEmbeddingGenerator<string, Embedding<float>> _embeddingGenerator;
+    private readonly IVectorStore _store;
+    private readonly IVectorStoreRecordCollection<int, ManualChunk> _collection;
+    private readonly string _collectionName = "ProductManuals";
+    private readonly string _vectorField = "Embedding";
+    public ProductManualService(IEmbeddingGenerator<string, Embedding<float>> embeddingGenerator, IVectorStore store)
+    {
+        _embeddingGenerator = embeddingGenerator;
+        _store = store;
+        _collection = _store.GetCollection<int, ManualChunk>(_collectionName);
+    }
+
+    public async Task InsertManualChunksAsync(IEnumerable<ManualChunk> manualChunks)
+    {
+
+        await _collection.CreateCollectionIfNotExistsAsync();
+
+        foreach (var chunk in manualChunks)
+        {
+            await _collection.UpsertAsync(chunk);
+        }
+    }
+
+    public async Task<VectorSearchResults<ManualChunk>> GetManualChunksAsync(string query, int? productId, int? limit = 5)
+    {
+        var queryEmbedding = await _embeddingGenerator.GenerateEmbeddingVectorAsync(query);
+
+        var searchOptions = new VectorSearchOptions
+        {
+            Top = limit ?? 1,
+            VectorPropertyName = _vectorField,
+            Filter = new VectorSearchFilter([
```

Comment:
@westey-m is this the proper use of Filter? Basically what I'm trying to do is filter on ProductId. I would expect to see the top 5 results, but I only get a single result which is not relevant to the query.

Reply:
Hi @luisquintanilla, yeah, that should work.  Although I think the following syntax is nicer:
```csharp
var filter = new VectorSearchFilter()
    .EqualTo(nameof(Glossary.Category), "External Definitions");
```

See https://learn.microsoft.com/en-us/semantic-kernel/concepts/vector-store-connectors/vector-search?pivots=programming-language-csharp#vectorsearchfilter

Also note that `VectorPropertyName` is not required if there is only one vector in the model.

Give me a shout later and I can help troubleshoot in person.